### ISSUE TITLE
Valgrind: Syscall param points to uninitialised byte(s) (bind)

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#1694]
Just something that came up while running RethinkDB under valgrind:

```
make DEBUG=1 VALGRIND=1 NO_TCMALLOC=1
valgrind [...]/rethinkdb --bind all
```

```
==26982== Memcheck, a memory error detector
==26982== Copyright (C) 2002-2011, and GNU GPL'd, by Julian Seward et al.
==26982== Using Valgrind-3.7.0 and LibVEX; rerun with -h for copyright info
==26982== Command: ./rethinkdb --bind all -o 23
==26982== 
info: Running rethinkdb foo (debug) (CLANG 3.2 (tags/RELEASE_32/final))...
info: Running on Linux 3.2.0-55-generic x86_64
info: Loading data from directory /home/ssd3/daniel/rethinkdb/build/debug_clang_valgrind_notcmalloc/rethinkdb_data
info: Our machine ID is b2c8794c-a56f-46c7-a43b-287eca32f84b
==26982== Thread 2:
==26982== Syscall param socketcall.bind(my_addr.sin6_scope_id) points to uninitialised byte(s)
==26982==    at 0x66071E7: bind (syscall-template.S:82)
==26982==    by 0x17FCC65: bind_ipv6_interface(int, int*, ip_address_t const&) (network.cc:759)
==26982==    by 0x17FC93B: linux_nonthrowing_tcp_listener_t::bind_sockets_internal(int*) (network.cc:790)
==26982==    by 0x17FBFE6: linux_nonthrowing_tcp_listener_t::bind_sockets() (network.cc:716)
==26982==    by 0x17FD53F: linux_tcp_bound_socket_t::linux_tcp_bound_socket_t(std::set<ip_address_t, std::less<ip_address_t>, std::allocator<ip_address_t> > const&, int) (network.cc:894)
==26982==    by 0x22921A7: connectivity_cluster_t::run_t::run_t(connectivity_cluster_t*, std::set<ip_address_t, std::less<ip_address_t>, std::allocator<ip_address_t> > const&, peer_address_t const&, int, message_handler_t*, int, heartbeat_manager_t*) (cluster.cc:112)
==26982==    by 0x1AFE612: do_serve(io_backender_t*, bool, base_path_t const&, metadata_persistence::cluster_persistent_file_t*, metadata_persistence::auth_persistent_file_t*, peer_address_set_t const&, service_address_ports_t, std::string, os_signal_cond_t*, boost::optional<std::string> const&) (serve.cc:153)
==26982==    by 0x1B0219F: serve(io_backender_t*, base_path_t const&, metadata_persistence::cluster_persistent_file_t*, metadata_persistence::auth_persistent_file_t*, peer_address_set_t const&, service_address_ports_t, std::string, os_signal_cond_t*, boost::optional<std::string> const&) (serve.cc:487)
==26982==    by 0x1ACE2C6: run_rethinkdb_serve(base_path_t const&, serve_info_t const&, file_direct_io_mode_t, int, uuid_u const*, cluster_semilattice_metadata_t const*, directory_lock_t*, bool*) (command_line.cc:759)
==26982==    by 0x1ACE61E: run_rethinkdb_porcelain(base_path_t const&, name_string_t const&, file_direct_io_mode_t, int, bool, serve_info_t const&, directory_lock_t*, bool*) (command_line.cc:783)
==26982==    by 0x1ADC3FE: _ZNSt5_BindIFPFvRK11base_path_tRK13name_string_t21file_direct_io_mode_tibRK12serve_info_tP16directory_lock_tPbES0_S3_S6_ibS7_SB_SC_EE6__callIvJEJLi0ELi1ELi2ELi3ELi4ELi5ELi6ELi7EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE (functional:1145)
==26982==    by 0x1ADC255: _ZNSt5_BindIFPFvRK11base_path_tRK13name_string_t21file_direct_io_mode_tibRK12serve_info_tP16directory_lock_tPbES0_S3_S6_ibS7_SB_SC_EEclIJEvEET0_DpOT_ (functional:1203)
==26982==  Address 0x432b9a0 is on thread 2's stack
==26982== 
info: Listening for intracluster connections on port 29038
==26982== Syscall param socketcall.bind(my_addr.sin6_scope_id) points to uninitialised byte(s)
==26982==    at 0x66071E7: bind (syscall-template.S:82)
==26982==    by 0x17FCC65: bind_ipv6_interface(int, int*, ip_address_t const&) (network.cc:759)
==26982==    by 0x17FC93B: linux_nonthrowing_tcp_listener_t::bind_sockets_internal(int*) (network.cc:790)
==26982==    by 0x17FBFE6: linux_nonthrowing_tcp_listener_t::bind_sockets() (network.cc:716)
==26982==    by 0x17FBB9F: linux_nonthrowing_tcp_listener_t::begin_listening() (network.cc:638)
==26982==    by 0x17FD711: linux_tcp_listener_t::linux_tcp_listener_t(std::set<ip_address_t, std::less<ip_address_t>, std::allocator<ip_address_t> > const&, int, boost::function<void ()(scoped_ptr_t<linux_tcp_conn_descriptor_t>&)> const&) (network.cc:907)
==26982==    by 0x2197E90: protob_server_t<ql::protob_t<Query>, Response, query2_server_t::context_t>::protob_server_t(std::set<ip_address_t, std::less<ip_address_t>, std::allocator<ip_address_t> > const&, int, boost::function<bool ()(ql::protob_t<Query>, Response*, query2_server_t::context_t*)>, Response (*)(ql::protob_t<Query>, std::string), boost::shared_ptr<semilattice_readwrite_view_t<auth_semilattice_metadata_t> >, protob_server_callback_mode_t) (protob.tcc:52)
==26982==    by 0x2196F36: protob_server_t<ql::protob_t<Query>, Response, query2_server_t::context_t>::protob_server_t(std::set<ip_address_t, std::less<ip_address_t>, std::allocator<ip_address_t> > const&, int, boost::function<bool ()(ql::protob_t<Query>, Response*, query2_server_t::context_t*)>, Response (*)(ql::protob_t<Query>, std::string), boost::shared_ptr<semilattice_readwrite_view_t<auth_semilattice_metadata_t> >, protob_server_callback_mode_t) (protob.hpp:57)
==26982==    by 0x21963D2: query2_server_t::query2_server_t(std::set<ip_address_t, std::less<ip_address_t>, std::allocator<ip_address_t> > const&, int, rdb_protocol_t::context_t*) (pb_server.cc:24)
==26982==    by 0x1B00DCE: do_serve(io_backender_t*, bool, base_path_t const&, metadata_persistence::cluster_persistent_file_t*, metadata_persistence::auth_persistent_file_t*, peer_address_set_t const&, service_address_ports_t, std::string, os_signal_cond_t*, boost::optional<std::string> const&) (serve.cc:381)
==26982==    by 0x1B0219F: serve(io_backender_t*, base_path_t const&, metadata_persistence::cluster_persistent_file_t*, metadata_persistence::auth_persistent_file_t*, peer_address_set_t const&, service_address_ports_t, std::string, os_signal_cond_t*, boost::optional<std::string> const&) (serve.cc:487)
==26982==    by 0x1ACE2C6: run_rethinkdb_serve(base_path_t const&, serve_info_t const&, file_direct_io_mode_t, int, uuid_u const*, cluster_semilattice_metadata_t const*, directory_lock_t*, bool*) (command_line.cc:759)
```

It might be a false alarm, but we should check it.

**Comment by Mandrilux** [comment]
Nice
